bluebridge-app.js

```diff
@@ -75,20 +79,35 @@ Polymer({
         });
     },
 
+    _firebaseStatusKnownChanged: function _firebaseStatusKnownChanged(newStatus) {
+        if (newStatus) {
+            console.log('[bluebridge-app] Firebase is ready for action');
+
+            var currentUser = firebase.auth().currentUser;
+
+            if (currentUser) {
+                this._setInternalFirebaseToken(currentUser);
```

Comment:
If the user has been to the site recently, but is signed-out then currentUser is defined with isAnonymous = true. 

If they are signed in, then then currentUser is also defined here with isAnonymous = false.

If they are in a completely new session (i.e. never been to the site, then currentUser is null). When currentUser is null we should use signInAnonymously.